PWGJE/EMCALJetTasks/macros/JetQA/scalePtHardHistos.py

```diff
@@ -220,7 +235,136 @@ def ScaleAllHistograms(obj, scaleFactor,f):
+        
+    if not cutLimitReached:
+      if foundAboveLimit and avg <= limit:
+        if cutIndex == -1:
+          cutIndex = index
+        nBinsBelowLimitAfterLimit += 1
+          
+      if nBinsBelowLimitAfterLimit != 0 and avg > limit:
+        # Reset
+        cutIndex = -1
+        nBinsBelowLimitAfterLimit = 0
+
+      if nBinsBelowLimitAfterLimit > nBinsThreshold:
+        cutLimitReached = True
+  # Do not perform removal here because then we miss values between the avg going below
+  # the limit and crossing the nBinsThreshold
+  
+  print("Hist checked: {0}, cut index: {1}".format(histToCheck.GetName(), cutIndex))
+  
+  # Use on both TH1 and TH2 since we don't start removing immediately, but instead only after the limit
+  if cutLimitReached:
+    print "Removing outliers"
+    # Check for values above which they should be removed by translating the global index
+    x = ctypes.c_int(0)
+    y = ctypes.c_int(0)
+    z = ctypes.c_int(0)
+    for index in range(0, hist.GetNcells()):
+      # Get the bin x, y, z from the global bin
+      hist.GetBinXYZ(index, x, y, z)
+      if y.value >= cutIndex:
+        if hist.GetBinContent(index) > 1e-3:
+          print("Cutting for index {}. y bin {}. Cut index: {}".format(index, y, cutIndex))
+          hist.SetBinContent(index, 0)
+          hist.SetBinError(index, 0)
+  else:
+    print "Hist {} did not have any outliers to cut".format(hist.GetName())
+  
+  # Check the mean and median
+  # Use another temporary hist
+  histToCheck = hist.ProjectionY("{histName}_projAfter".format(histName = hist.GetName()))
+  
+  (postMean, postMedian) = GetHistMeanAndMedian(histToCheck)
+  print "Pre  outliers removal mean: {}, median: {}".format(preMean, preMedian)
+  print "Post outliers removal mean: {}, median: {}".format(postMean, postMedian)
+
+  #plotHist(hist, "./{}_{}.pdf".format(hist.GetName(), "After"))
+  #plotHist(histToCheck, "./{}.pdf".format(histToCheck.GetName()), "hist E", True)
+
+def GetHistMeanAndMedian(hist):
+  # Median
+  # See: https://root-forum.cern.ch/t/median-of-histogram/7626/5
+  x = ctypes.c_double(0)
+  q = ctypes.c_double(0.5)
+  # Apparently needed to be safe(?)
+  hist.ComputeIntegral()
+  hist.GetQuantiles(1, x, q)
+    
+  mean = hist.GetMean()
+  return (mean, x.value)
+
+def MovingAverage(hist, index, numberOfCountsBelowIndex = 0, numberOfCountsAboveIndex = 2):
+  """
+  # [-2, 2] includes -2, -1, 0, 1, 2
+  """
+  # Check inputs
+  if numberOfCountsBelowIndex < 0 or numberOfCountsAboveIndex < 0:
+    print "Moving average number of counts above or below must be >= 0. Please check the values!"
+          
+  count = 0.
+  average = 0.
+  for i in range(index - numberOfCountsBelowIndex, index + numberOfCountsAboveIndex + 1):
+    # Avoid going over histogram limits
+    if i < 0 or i >= hist.GetNcells():
+      continue
+    # TEMP - Hack due to entries in pt hard value = 1 for large pt hard bins for no apparent reason
+    if i == 1:
+      continue
```

Comment:
You probably should remove this temp hack. I've addressed it in the embedding helper, so it shouldn't be an issue anymore